devito/dle/transformer.py

```diff
@@ -1,82 +1,55 @@
-                 transformations; in the latter case, the specified transformations
-                 are composed. We use the following convention: ::
-
-                    * [S]: a pre-defined sequence of transformations.
-                    * [T]: a single transformation.
-                    * [sT]: a single "speculative" transformation; that is, a
-                            transformation that might increase, or even decrease,
-                            performance.
-
-                 The keywords usable in/as ``mode`` are: ::
-
-                    * 'noop': Do nothing -- [S]
-                    * 'basic': Apply all of the available legal transformations
-                               that are most likely to increase performance (ie, all
-                               [T] listed below), except for loop blocking -- [S]
-                    * 'advanced': Like 'basic', but also switches on loop blocking -- [S]
-                    * '3D-advanced': Like 'advanced', but apply 3D loop blocking
-                                     if there are at least the perfectly nested
-                                     parallel iteration spaces -- [S]
-                    * 'speculative': Apply all of the 'advanced' transformations,
-                                     plus other transformations that might increase
-                                     (or possibly decrease) performance -- [S]
-                    * 'fission': Apply loop fission -- [T]
-                    * 'blocking': Apply loop blocking -- [T]
-                    * 'split': Identify and split elemental functions -- [T]
-                    * 'padding': Introduce "shadow" arrays, padded to the nearest
-                                 multiple of the vector length, to maximize data
-                                 alignment -- [T]
-                    * 'simd': Add pragmas to trigger compiler auto-vectorization -- [T]
-                    * 'ntstores': Add pragmas to issue nontemporal stores -- [sT]
-
-                 If ``mode`` is a tuple, the last entry may be used to provide optional
-                 arguments to the DLE transformations. Accepted key-value pairs are: ::
-
-                    * 'blockshape': A tuple representing the shape of a block created
-                                    by loop blocking.
-                    * 'blockinner': By default, loop blocking is not applied to the
-                                    innermost dimension of an Iteration/Expression tree
-                                    to maximize the chances of SIMD vectorization. To
-                                    force the blocking of this loop, the ``blockinner``
-                                    flag should be set to True.
-                    * 'openmp': True to emit OpenMP code, False otherwise.
+    :param mode: Drive the tree transformation. ``mode`` is a string indicating
+                 a certain optimization pipeline. The following values are accepted: ::
```

Comment:
So I take creating custom optimisation sequence is no longer available, and users have to rely on the preset sequences?